### PR TITLE
CRM GYM Hibernate Task

### PR DESCRIPTION
**Pull request by luismateoh**
Implemented user registration, login, profile management, training session management, and activation/deactivation of users.

### CODE REVIEW

File: src/main/java/com/luismateoh/gymcrm/dto/TrainerDTO.java [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+
+    @NotNull(message = "User cannot be null")
+    private UserDTO user;
+
+    @NotNull(message = "Specialization cannot be null")
+    private TrainingTypeDTO specialization;
+
+    @Override
+    public String toString() {
+        return "Trainer -> " + "ID: " + id + ", Specialization: " + specialization.getTrainingTypeName() + "\n" + user + "\n";
```
Comment (jugonzalez-clgx) [inline src/main/java/com/luismateoh/gymcrm/dto/TrainerDTO.java:20]: String interpolation is  a better approach. Sample String.format("String %s in %s with some %s examples.", first, second, second)

File: src/main/java/com/luismateoh/gymcrm/dao/TrainingTypeDao.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@Repository
+public class TrainingTypeDao extends GenericDaoImpl<TrainingType, Long> {
+
+    public TrainingTypeDao(SessionFactory sessionFactory) {
+        super(TrainingType.class, sessionFactory);
+    }
+
+    public TrainingType findByName(String name) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<TrainingType> query = session.createQuery("FROM TrainingType WHERE trainingTypeName = :name", TrainingType.class);
```
Comment (jugonzalez-clgx) [inline src/main/java/com/luismateoh/gymcrm/dao/TrainingTypeDao.java:18]: It's good practice to have queries isolated in other file

**Review by jugonzalez-clgx — APPROVED** [review]
Well done, minor suggestions to be used in upcoming tasks

**Comment by luismateoh** [comment]
Thanks for revision and suggestions!

**Pull request merged**